frontend/apps/reader/modules/readerpaging.lua

```diff
@@ -847,61 +866,88 @@ function ReaderPaging:onGotoPageRel(diff)
         -- negative x panning if writing direction is right to left
         local direction = self.ui.document.configurable.writing_direction
         x_pan_off = self.visible_area.w * diff * (direction == 1 and -1 or 1)
-    elseif self.zoom_mode:find("column") then
-        -- zoom mode for two-column navigation
 
-        y_pan_off = self.visible_area.h * diff
-        y_pan_off = Math.roundAwayFromZero(y_pan_off)
-        new_va.x = Math.roundAwayFromZero(self.visible_area.x)
-        new_va.y = Math.roundAwayFromZero(self.visible_area.y+y_pan_off)
-        -- intra-column navigation (vertical), this is the default behavior
-        -- if we do not reach the end of a column
+    elseif self.zoom_mode == "pan" or self.zoom_mode == "column" then
+        -- pan zoom mode
+
+        local old_va = self.visible_area
+        local x, y, w, h = "x", "y", "w", "h"
+        local h_progress = 1 - self.zoom_pan_h_overlap / 100
+        local v_progress = 1 - self.zoom_pan_v_overlap / 100
+        local x_diff = diff
+        local y_diff = diff
+
+        if self.zoom_pan_direction_vertical then  -- invert axes
+            y, x, h, w = x, y, w, h
+            h_progress, v_progress = v_progress, h_progress
+            if self.ui.document.configurable.writing_direction == 1 then
```

Comment:
You shouldn't use this. As I said it's for reflow. What you want is the `inverse_reading_order` setting.

Reply:
We already have a global (overridable in docsetting I think) setting named `inverse_reading_order`, which just swap the tap zone and swipe direction for page turning (and a few minor other things).
I may be wrong as I have not really read this PR, but I think the one you want is a bit orthogonal to that (and so, should have another name):
@yparitcher reads RTL hebrew pages, but still want to use english reading order - so he expects tapping on the right half of the screen will go page forward, so actually with this PR showing the next pan area.
So, looks like we need 3 RTL-related settings:
- inverse reading order, for tap and swipes
- a new one for zoom pan RTL zone ordering
- the current one (writting direction) for Reflow mode

Do I get that right ? In which case, try to find another name than `inverse_reading_order` for the new one, so there's no confusion. (If not, and it should re-use the normal inverse_reading_order, it shouldn't be in document.configurable.)

Reply:
The fact is:

- I'd need something in `document.configurable`, to avoid multiple "readSettings";
- l. 855, for `*height` zoom modes, `self.ui.document.configurable.writing_direction` is used.

So:

- should I come back to `self.ui.document.configurable.writing_direction` ?
- if I create a new one, should I change l. 855 too ?

Reply:
I certainly wasn't referring to any kind of new setting under configurable. ;-)

Things can be rewritten if necessary, which could theoretically include repurposing the reflow stuff, but please do not add any new settings related to this since there are more than enough already.

Reply:
> So, looks like we need 3 RTL-related settings:

The reflow and zoom pan RTL can be the same setting, but the separate reflow and inverse are a thorn in my eye tbh.

Reply:
> I'd need something in document.configurable, to avoid multiple "readSettings";

You either don't quite mean what you wrote or that's fallacious. :-)

> l. 855, for *height zoom modes, self.ui.document.configurable.writing_direction is used.

There's a bug in the program then, because without activating reflow you can't change that setting. That's implicitly part of what I was referring to.

Reply:
> I'd need something in document.configurable, to avoid multiple "readSettings";

I think the only reason you would need document.configurable is because that's required for anything bottom menu.
(readSettings costs nothing: it just picks a value from a table by key).

If reflow (that uses the existing `configurable.writing_direction `) mode, and this new pan zoom mode can't be active at the same time, I guess that then, yes, it might be a common setting. After all, it's a property of the document that it is written in LTR or RTL direction, so that generic name could be used by different code to do their own thing with this document characteristic.
(I guess a document could have pages in LTR and others in RTL - but we can ignore this case I guess :)

Still need some input from RTL readers :) @yparitcher ?

Reply:
> If reflow (that uses the existing configurable.writing_direction ) mode, and this new pan zoom mode can't be active at the same time, I guess that then, yes, it might be a common setting. After all, it's a property of the document that it is written in LTR or RTL direction, so that generic name could be used by different code to do their own thing with this document characteristic.

Indeed, the restriction could simply be removed from that setting so that it can always be changed — not sure what to do about TBRTL though.

Reply:
> The reflow and zoom pan RTL can be the same setting, but the separate reflow and inverse are a thorn in my eye tbh.

What I suggest for now, in order to keep issues separated:

- I go back to `self.ui.document.configurable.writing_direction`, like `*height`, but make it configurable in `Pan zoom settings`;
- once this PR is merged, I'm willing to review the `self.ui.document.configurable.writing_direction` - `inverse_reading_order` in a new PR (IMHO, one should imply the other, at least at the document level).

Reply:
> self.ui.document.configurable.writing_direction, like *height, but make it configurable in Pan zoom settings;

The alternative should pretty much just be removing a condition? I don't understand the rationale here. I suppose it doesn't matter too much as long as you use the same strings.

Reply:
> > I'd need something in document.configurable, to avoid multiple "readSettings";
> 
> You either don't quite mean what you wrote or that's fallacious. :-)

I was referring to https://github.com/koreader/koreader/pull/6885#issuecomment-729289403:

> it might be a good idea to store the runtime settings in self.ui.document.configurable, so as not to have to load them in many places? this might also prevent stale values in the future.

If I didn't understand well, may you please point me to the best practice ?

Reply:
@poire-z 
> I think the only reason you would need document.configurable is because that's required for anything bottom menu.
> (readSettings costs nothing: it just picks a value from a table by key).

Just to make things clear before I revert the changes I did: if I have a settings that could change while the document is loaded (especially `zoom_factor`), should it be in `document.configurable`, or should I read/save it directly with `readSetting/saveSetting` ? In fact, I can't really understand the purpose of `configurable`.

Reply:
I believe (I rarely look at these parts of the code) that `configurable` is where the bottom menu stores/handles its settings, a common place that is shared by all the things involved (ReaderConfig, ConfigDialog, koptoptions, creoptions...) - there's a lot of interactions between all these, so they all go through reading/updating `configurable` - and only when saving a book, these things are stored in docsetting or G_reader_settings (which might never be done explicitely - haven't verified this in the code though :) 
(i.e. when you toggle a toggle, there's lot of generic code in togglewidget/configdialog that don't save anything explicitely - and we have some unresolved issues where toggling a bottom setting via a gesture does not have the bottom/toggle state updated, etc...)

It's only in the top menu where things are simpler that we can readSetting/saveSetting at will.

So, unless you want to dig deeper in how the bottom menu works (which I don't forbid you to do :) , it's less work to just mimic/copy how some existing bottom setting is handled.
(If you're moving stuff from the top menu to the bottom menu, I feel there's a bit more changed involved than just what appears to the eye :/)

Reply:
And all this might not always be clean. I believe there are some settings that are handled and saved by the bottom menu in one name, but are also saved with another name by the reader module that get the event send by the bottom menu :/ It's possible early, the bottom menu was only doing simple things and this whole handling was avoiding have to do anything in the reader module - but since then, we have more stuff handled in the events - which still having to keep the bottom menu crap so it can still works by having/handling its own vision of the state of settings.

Reply:
@poire-z 
You got it right, the document direction and the page turning direction are not always the same. this is why currently `inverse_reading_order` is called "Invert page turn taps and swipes" in the UI **not** RTL document.

@jperon
what i meant was to try to have one source of truth instead of loading the settings all over the place, (this may not be possible) I Didn't intend `self.ui.document.configurable` specifically it was an example. sorry.

For the menu, it might be easier for the user to combine the 3 checkboxes "column mode, RTL, Bottom to Top" into one item called Pan direction/mode/order with 8 options LRTB/TBLR/RLTB/TBRL/LRBT/BTLR/RLBT/BTRL preferably with visual arrow icons as in the screenshot in https://github.com/koreader/koreader/issues/3685#issuecomment-366540785 (i will look for some) that will be self understood. This would fit very much in the margin bottom menu.

Reply:
> what i meant was to try to have one source of truth instead of loading the settings all over the place

That I can get behind. ;-)

Reply:
So I ended not `configurable` any more, except for `configurable.writing_direction` that I use to set `right-to-left`. The settings are defined and saved from `ReaderZooming`; they are read by the mean of `readSetting` to local variables in `ReaderPaging:onGotoPageRel`

Reply:
While testing more, I saw that, not using `configurable.writing_direction`, I couldn't do less than:

- read and save settings in `ReaderZooming` (as they are defined from there);
- read settings from `ReaderPaging`:
    - at load, to have them available;
    - each time a `ZoomPanUpdate` event occurs, to update the setting(s) that get changed and update the view at once.